packages/dds/migration-shim/src/shimDeltaConnection.ts

```diff
@@ -23,47 +23,48 @@ import { type IShimDeltaHandler } from "./types";
  * modifications on the current ChannelDeltaConnection.
  */
 export class ShimDeltaConnection implements IDeltaConnection {
-	public constructor(private readonly deltaConnection: IDeltaConnection) {}
-	private _handler: IShimDeltaHandler | undefined;
-	private get handler(): IShimDeltaHandler {
-		assert(this._handler !== undefined, "Missing delta handler");
-		return this._handler;
-	}
+	public constructor(
+		private readonly deltaConnection: IDeltaConnection,
+		private readonly shimDeltaHandler: IShimDeltaHandler,
+	) {}
+	private isShimDeltaHandlerAttachedToConnection = false;
 
 	public get connected(): boolean {
 		return this.deltaConnection.connected;
 	}
 
 	// Should we be adding some metadata here?
 	public submit(messageContent: unknown, localOpMetadata: unknown): void {
-		// The expectation is to currently add some form of V2 stamp to the metadata here that can eventually be
-		// ignored by a SharedObject V2
-		// TODO: stamp messageContent with V2 metadata
+		// TODO: stamp messageContent with V2 metadata - this is not the final implementation
+		this.shimDeltaHandler.stampV2Ops(messageContent);
```

Comment:
I think the stamping should probably happen internally to the shim?  Since the MigrationShim will not always be stamping v2 but instead will need to make a decision about how to stamp.

Reply:
OK, I'll extract out the v2 stamping. We can do that later